doc/jvm/xgboost_spark_migration.rst

```diff
@@ -0,0 +1,140 @@
+######################################################
+Migration Guide: How to migrate from XGBoost Spark 3.x
+######################################################
+
+XGBoost Spark underwent significant modifications beginning with version 3.0,
+which may cause compatibility issues with existing user code.
+
+This guide will walk you through the process of updating your code to ensure
+it's compatible with XGBoost Spark 3.0 and later versions.
+
+**********************
+XGBoost Spark Packages
+**********************
+
+XGBoost Spark 3.0 introduced a single uber package named xgboost-spark_2.12-3.0.0.jar, which bundles 
+both xgboost4j and xgboost4j-spark. This means you can now simply use `xgboost-spark`` for your application.
+
+.. code-block:: xml
+
+  <dependency>
+      <groupId>ml.dmlc</groupId>
+      <artifactId>xgboost-spark_${scala.binary.version}</artifactId>
+      <version>3.0.0</version>
+  </dependency>
+
+When submitting the XGBoost application to the Spark cluster, you only need to specify the single `xgboost-spark` package.
+
+.. code-block:: bash
+
+  spark-submit \
+    --jars xgboost-spark_2.12-3.0.0.jar \
+    ... \
+
+**************
+XGBoost Ranking
+**************
+
+The ability to handle ranking problems using XGBoostRegressor has been discontinued.
+As an alternative, we have introduced XGBoostRanker, which is specifically designed
+to support ranking algorithms.
+
+.. code-block:: scala
+  
+  // before 3.0
+  val regressor = new XGBoostRegressor().setObjective("rank:ndcg")
+
+  // after 3.0
+  val ranker = new XGBoostRanker()
+
+******************************
+XGBoost Constructor Parameters
+******************************
+
+XGBoost Spark now categorizes parameters into two groups: XGBoost-Spark parameters and XGBoost parameters.
```

Comment:
Where can I find which parameter belongs to the spark parameter group?

Reply:
no where,  user can use setter functions.

Reply:
We don't have a page for that, but I think the API docs can have these.